doc/main/reference/try_put_and_wait.rst

```diff
@@ -0,0 +1,324 @@
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true`` if the input was accepted, ``false`` otherwise.
+
+.. code:: cpp
+
+    template <typename T>
+    bool overwrite_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Stores ``input`` in the internal single item buffer and broadcasts it to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. caution::
+
+    Since the input element is not retrieved from ``overwrite_node`` once accepted by the successor, the element should be
+    retrieved by explicitly calling the ``clear()`` method or by overwriting by another element to prevent
+    ``try_put_and_wait`` from endless waiting.
+
+.. code:: cpp
+
+    template <typename T>
+    bool write_once_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Stores ``input`` in the internal single item buffer if it does not contain a valid value already.
+If a new value is set, the node broadcast it to all successors.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true`` for the first time after construction or a call to ``clear()``.
+
+.. caution::
+
+    Since the input element is not retrieved from the ``write_once_node`` once accepted by the successor, the element should
+    be retrieved by explicitly calling the ``clear()`` method to prevent ``try_put_and_wait`` from endless waiting.
+
+.. code:: cpp
+
+    template <typename T>
+    bool buffer_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the set of items managed by the node and tries forwarding it to a successor.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. code:: cpp
+
+    template <typename T>
+    bool queue_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the set of items managed by the node, and tries forwarding the least recently added item
+to a successor.
+
+Waits for the completion of the ``input`` in the Flow Graph meaning all tasks in each graph node related to ``input``
+are exetutted and no buffered objects related to ``input`` in any buffer in the graph.
+
+**Returns**: ``true``.
+
+.. code:: cpp
+
+    template <typename T, typename Compare>
+    bool priority_queue_node<T>::try_put_and_wait(const T& input)
+
+**Effects**: Adds ``input`` to the ``priority_queue_node`` and tries forwarding to a successor the item with the largest
```

Comment:
Adds input to the priority_queue_node and attempts to forward the item with the highest priority among all items added to the node but not yet forwarded to the successors.

Reply:
Fixed